libtenzir/builtins/functions/config.cpp

```diff
@@ -28,12 +28,18 @@ class plugin final : public virtual function_plugin {
     config_ = global_config;
     // This one's very noisy and not particularly user-facing, so we hide it.
     config_.erase("caf");
-    // This one really shouldn't be exposed.
     if (auto* tenzir = get_if<record>(&config_, "tenzir")) {
+      // Remove secrets.
       tenzir->erase("secrets");
       tenzir->erase("token");
+      // This one's an implementation detail of the test runner, and if we don't
+      // delete it then it may unexpectedly show up when using the `config()`
+      // function in tests.
+      tenzir->erase("disable-banner");
+      // This one's an implementation detail of the Nix-created Docker image.
+      tenzir->erase("runtime-prefix");
```

Comment:
I think we should have a general "testing" mode that enables such things. Not saying we should remove `TENZIR_DISABLE_BANNER`, that's useful in general.

Reply:
I agree, but that's out of scope for this PR, I think.